File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Popup;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
    }

    public class MyDateEdit: DateEdit
    {
        
        public MyDateEdit()
        {
            
        }
        protected override DevExpress.XtraEditors.Popup.PopupBaseForm CreatePopupForm()
        {
            return new MyVistaPopupDateEditForm(this);
        }
    }

    public class MyVistaPopupDateEditForm : VistaPopupDateEditForm
    {
        public MyVistaPopupDateEditForm(DateEdit ownerEdit)
            : base(ownerEdit)
        {
            
        }
        protected override DateEditCalendar CreateCalendar()
        {
            VistaDateEditCalendar c = new MyVistaDateEditCalendar(OwnerEdit.Properties, OwnerEdit.EditValue);
            c.OkClick += new EventHandler(OnOkClick);
            return c;
        }
        protected virtual void OnOkClick(object sender, EventArgs e)
        {
            if (OwnerEdit != null)
                OwnerEdit.ClosePopup();
        }


    }

    public class MyVistaDateEditCalendar : VistaDateEditCalendar
    {
        public MyVistaDateEditCalendar(DevExpress.XtraEditors.Repository.RepositoryItemDateEdit item, object editDate)
            : base(item, editDate)
        {
            
        }

        public override Size CalcBestSize(Graphics g)
        {
            return new Size(VistaCalendar.ClockTextRect.Width + 40, VistaCalendar.ClockRect.Height + VistaCalendar.ClockTextRect.Height  +4 *VistaCalendar.IndentFromTopToClock);
            
            }

        protected override DevExpress.XtraEditors.ViewInfo.DateEditInfoArgs CreateInfoArgs()
        {
            return new MyVistaDateEditInfoArgs(this);
        }
        
   
    }

    public class MyVistaDateEditInfoArgs : VistaDateEditInfoArgs
    {
        public MyVistaDateEditInfoArgs(DateEditCalendarBase calendar)
            : base(calendar)
        {
            
        }

        public override void CalcLayout(Rectangle contentBounds, DevExpress.XtraEditors.Calendar.CalendarBasePainter painter)
        {
            contentBounds.Width = 10;
            contentBounds.Y= -40;
            base.CalcLayout(contentBounds, painter);
            Rectangle buttonBounds = ClockTextRect;
            buttonBounds.Y += 30;
            buttonBounds.Width = ClockTextRect.Width / 2 - 5;
            OkRect = buttonBounds;
            OkButtonRect = GetButtonRect(OkRect);
            buttonBounds.Offset(buttonBounds.Width + 5, 0);
            CancelRect = buttonBounds;
            CancelButtonRect = GetButtonRect(buttonBounds);
        }


        
    }


}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.Popup
Imports DevExpress.XtraEditors.Controls
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
	End Class

	Public Class MyDateEdit
		Inherits DateEdit

		Public Sub New()

		End Sub
		Protected Overrides Function CreatePopupForm() As DevExpress.XtraEditors.Popup.PopupBaseForm
			Return New MyVistaPopupDateEditForm(Me)
		End Function
	End Class

	Public Class MyVistaPopupDateEditForm
		Inherits VistaPopupDateEditForm
		Public Sub New(ByVal ownerEdit As DateEdit)
			MyBase.New(ownerEdit)

		End Sub
		Protected Overrides Function CreateCalendar() As DateEditCalendar
			Dim c As VistaDateEditCalendar = New MyVistaDateEditCalendar(OwnerEdit.Properties, OwnerEdit.EditValue)
			AddHandler c.OkClick, AddressOf OnOkClick
			Return c
		End Function
		Protected Overridable Overloads Sub OnOkClick(ByVal sender As Object, ByVal e As EventArgs)
			If OwnerEdit IsNot Nothing Then
				OwnerEdit.ClosePopup()
			End If
		End Sub


	End Class

	Public Class MyVistaDateEditCalendar
		Inherits VistaDateEditCalendar
		Public Sub New(ByVal item As DevExpress.XtraEditors.Repository.RepositoryItemDateEdit, ByVal editDate As Object)
			MyBase.New(item, editDate)

		End Sub

		Public Overrides Function CalcBestSize(ByVal g As Graphics) As Size
			Return New Size(VistaCalendar.ClockTextRect.Width + 40, VistaCalendar.ClockRect.Height + VistaCalendar.ClockTextRect.Height +4 *VistaCalendar.IndentFromTopToClock)

		End Function

		Protected Overrides Function CreateInfoArgs() As DevExpress.XtraEditors.ViewInfo.DateEditInfoArgs
			Return New MyVistaDateEditInfoArgs(Me)
		End Function


	End Class

	Public Class MyVistaDateEditInfoArgs
		Inherits VistaDateEditInfoArgs
		Public Sub New(ByVal calendar As DateEditCalendarBase)
			MyBase.New(calendar)

		End Sub

		Public Overrides Sub CalcLayout(ByVal contentBounds As Rectangle, ByVal painter As DevExpress.XtraEditors.Calendar.CalendarBasePainter)
			contentBounds.Width = 10
			contentBounds.Y= -40
			MyBase.CalcLayout(contentBounds, painter)
			Dim buttonBounds As Rectangle = ClockTextRect
			buttonBounds.Y += 30
			buttonBounds.Width = ClockTextRect.Width \ 2 - 5
			OkRect = buttonBounds
			OkButtonRect = GetButtonRect(OkRect)
			buttonBounds.Offset(buttonBounds.Width + 5, 0)
			CancelRect = buttonBounds
			CancelButtonRect = GetButtonRect(buttonBounds)
		End Sub



	End Class


End Namespace